deprecating supports_distributed_data causes tons of warning

Instantiation  of related vector classes now trigger warning like
```
In file included from /Users/qiaol/devel/NSolver/source/solver/solve.cpp:8:
In file included from /Users/qiaol/devel/NSolver/header/NSolver/solver/NSolver.h:98:
In file included from /Users/qiaol/devel/NSolver/header/NSolver/types.h:12:
In file included from /Users/qiaol/Library/deal.II/include/deal.II/lac/generic_linear_algebra.h:23:
In file included from /Users/qiaol/Library/deal.II/include/deal.II/lac/block_vector.h:23:
/Users/qiaol/Library/deal.II/include/deal.II/lac/block_vector_base.h:576:79: warning: 'supports_distributed_data' is deprecated [-Wdeprecated-declarations]
  static const bool supports_distributed_data DEAL_II_DEPRECATED = BlockType::supports_distributed_data;
                                                                              ^
/Users/qiaol/Library/deal.II/include/deal.II/lac/petsc_parallel_block_vector.h:62:32: note: in instantiation of template class 'dealii::BlockVectorBase<dealii::PETScWrappers::MPI::Vector>' requested here
    class BlockVector : public BlockVectorBase<Vector>
                               ^
/Users/qiaol/Library/deal.II/include/deal.II/lac/petsc_parallel_vector.h:179:25: note: 'supports_distributed_data' has been explicitly marked deprecated here
      static const bool supports_distributed_data DEAL_II_DEPRECATED = true;
```

--- a/include/deal.II/lac/block_vector_base.h
+++ b/include/deal.II/lac/block_vector_base.h
@@ -573,7 +573,7 @@ public:
    * @deprecated instead of using this variable, please use the type trait value
    * <code>is_serial_vector< VectorType >::value</code>
    */
-  static const bool supports_distributed_data DEAL_II_DEPRECATED = BlockType::supports_distributed_data;
+  static const bool supports_distributed_data DEAL_II_DEPRECATED = !is_serial_vector<BlockType>::value;
 
   /**
    * Default constructor.

